packages/server/modules/cross-server-sync/services/commit.ts

```diff
@@ -362,318 +370,305 @@ const cleanViewerState = (
-    }
+    // Iterate over all objects and download them into the DB
+    const totalObjectCount = (sourceCommit.totalChildrenCount || 0) + 1
+    const batchSize = 50
+    let batchPromises: Promise<unknown>[] = []
+    let processedObjectCount = 1
+
+    for await (const obj of objectLoader.getObjectIterator()) {
+      const typedObj = obj as ObjectLoaderObject
+      const work = async () => {
+        const id = `${obj.id} - ${processedObjectCount++}/${totalObjectCount}`
+        logger.debug(`Processing ${id}...`)
+        await retry(
+          () =>
+            Promise.race([
+              deps.createNewObject(typedObj, targetStreamId, { logger }),
+              timeoutAt(10 * 1000, `Object create timed out! - ${id}`)
+            ]),
+          3
+        )
+        logger.debug(`Processed! ${id}`)
+      }
 
-    batchPromises.push(work())
-    if (batchPromises.length >= batchSize) {
-      await Promise.all(batchPromises)
-      batchPromises = []
+      batchPromises.push(work())
+      if (batchPromises.length >= batchSize) {
+        await Promise.all(batchPromises)
+        batchPromises = []
+      }
     }
+
+    // If any remaining promises - await them
+    await Promise.all(batchPromises)
   }
 
-  // If any remaining promises - await them
-  await Promise.all(batchPromises)
-}
+type DownloadCommitDeps = GetLocalResourcesDeps &
+  SaveNewCommitDeps &
+  CreateNewObjectDeps &
+  SaveNewThreadsDeps
 
 /**
  * Downloads a commit/version (both FE1 and FE2 supported) from an external Speckle server instance
  */
-export const downloadCommit = async (
-  argv: {
-    /**
-     * A FE1 commit URL or an FE2 model/version URL
-     */
-    commitUrl: string
-    /**
-     * ID of the local stream that should receive the commit
-     */
-    targetStreamId: string
-    /**
-     * Stream branch that should receive the commit. Defaults to 'main'
-     */
-    branchName?: string
-    /**
-     * Specify if target commit is private
-     */
-    token?: string
-    /**
-     * Specify if you want comments to be pulled in also
-     */
-    commentAuthorId?: string
-
-    /**
-     * Specify if you want to sync in automation statuses. If set to true, the Project.version
-     * query will be used, which might not be supported if you're targetting an old server instance.
-     */
-    loadAutomations?: boolean
-  },
-  options?: Partial<{
-    logger: Logger
-  }>
-) => {
-  const {
-    commitUrl,
-    targetStreamId,
-    branchName = 'main',
-    token,
-    commentAuthorId
-  } = argv
-  const { logger = crossServerSyncLogger } = options || {}
-
-  logger.debug(`Commit/version download started at: ${new Date().toISOString()}`)
-
-  const localResources = await getLocalResources(
-    targetStreamId,
-    branchName,
-    commentAuthorId
-  )
-  logger.debug(
-    `Using local branch ${branchName} of stream ${targetStreamId} to dump the incoming commit`
-  )
+export const downloadCommitFactory =
```

Comment:
I made all DB operations injectable, but left out all remote server access as is, cause realistically I don't see why we'd want to mock it out (it already supports running queries against any speckle server in and of itself) and mocking out stuff like the ObjectLoader could be complicated